com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/BobProjectProperties.java

```diff
@@ -333,15 +526,37 @@ static private Map<String, Map<String, String>> doLoad(InputStream in, KeyValueF
                 boolean valid = true;
                 if (sep != -1) {
                     try {
-                        String key = line.substring(0, sep).trim();
+                        String fullKey = line.substring(0, sep).trim();
                         String value = line.substring(sep + 1).trim();
-                        if (passFunc != null) {
-                            StringKeyValue keyVal = new StringKeyValue(key, value);
-                            if (passFunc.filter(keyVal)) {
-                                propGroup.put(keyVal.key, value);
+                        if(fullKey.contains(".")) {
+                            String[] keyAndAttr = fullKey.split("\\.");
+                            String key = keyAndAttr[0];
+                            ProjectProperty prop = propGroup.get(key);
+                            if (prop == null) {
+                                prop = new ProjectProperty();
+                            }
+                            prop.setAttribute(keyAndAttr[1], value);
+                            propGroup.put(key, prop);
+                        }
+                        else if (!isMeta && fullKey.contains("#")) {
```

Comment:
universal parsing of array values (with index after `#`)

Reply:
I'm not sure about this whole if-statement.
If the key contains a `.`, then we'll never get here?

Reply:
yes, you right
you can't specify attribute for an element in array
so, if it's attribute (it has `.` in it) then we parse it as an attribute